# Задача 2. Глобальная оптимизация

$$ f(x) = sin(x / 5) * exp(x / 10) + 5 * exp(-x / 2) $$

$$ промежуток [1,30]$$

In [17]:
import scipy
from scipy import optimize
import math

In [18]:
# исходная функция, вычисление значений
def our_func(x):
    return (math.sin(x/5.0))*(math.exp(x/10.0)) + 5*(math.exp(-x/2.0))

In [19]:
res = scipy.optimize.differential_evolution(our_func, [(1,30)]) 
# список кортежей - это список областей определения для каждого аргумента функции
print res

     fun: -11.898894665981299
     jac: array([0.])
 message: 'Optimization terminated successfully.'
    nfev: 98L
     nit: 5
 success: True
       x: array([25.88019331])


In [20]:
fun_value = res.fun
print res.x, res.fun

[25.88019331] -11.898894666


## Ответ:
$$ x = 25.88019326$$
$$fun = -11.898894666 $$

Данный результат корректный и соотносятся с графиком исходной функции.
    Количество вычислений для дифференциального генетического алгоритма значительно выше (nfev = 98) чем у BFGS (nfev = 21), как и колличество совершенных итераци. Это соотносится с особенностями работы обоих алгоритмов.

#### Запись результатов в файл

In [22]:

with open('task_srcond_answer.txt', 'w') as func_value:
    func_value.write(str(fun_value))